kinto/authorization.py

```diff
@@ -145,4 +145,7 @@ def get_bound_permissions(self, *args, **kwargs):
 
 
 class RouteFactory(core_authorization.RouteFactory):
-    pass
+    def __init__(self, request):
+        super(RouteFactory, self).__init__(request)
+        if self.on_collection and self.resource_name == 'bucket':
+            self.force_empty_list = True
```

Comment:
This attribute is not defined if the condition is not met.

Instead, you could define another RouteFactory (e.g.  `BucketRouteFactory` with a class attribute like `allow_empty_list`)